app/routes/bdb/components/AddSemester.tsx

```diff
@@ -62,49 +75,41 @@ const AddSemester = () => {
 
   const [submit, setSubmit] = useState(false);
   const [foundSemesterStatus, setFoundSemesterStatus] = useState<{
-    semester: string;
+    semester: Semester;
     year: number;
-  } | null>(null);
-
-  const onSubmit = ({
-    year,
-    semester,
-    contract,
-    semesterStatus,
-  }: SemesterStatusEntity) => {
+  }>();
+
+  if (!company) {
+    return <LoadingIndicator loading />;
+  }
+
+  const onSubmit = ({ year, semester, semesterStatus }: FormValues) => {
+    setFoundSemesterStatus(undefined);
+
     const contactedStatus = semesterStatus.contactedStatus;
 
     if (!submit) return;
```

Comment:
Oh, it's horrible but out of scope to fix in this PR. The buttons inside the semester-status picker thing are all submit-buttons, so without the submit state it would automatically submit the entire form when selecting a semester status. It's been like this for years.